### PR TITLE
bpo-35602: Make sure the transport is always closed in SelectorEventLoopUnixSockSendfileTests

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue35602

### CODE REVIEW

File: Lib/test/test_asyncio/test_unix_events.py [hunk 1]
```diff
@@ -504,12 +504,9 @@ def prepare(self):
             lambda: proto, sock=srv_sock))
         self.run_loop(self.loop.sock_connect(sock, (support.HOST, port)))
 
-        def cleanup():
-            if proto.transport is not None:
-                # can be None if the task was cancelled before
-                # connection_made callback
-                proto.transport.close()
-                self.run_loop(proto.wait_closed())
+        def cleanup(transport=proto.transport):
```
Comment (asvetlov) [inline Lib/test/test_asyncio/test_unix_events.py]: Looks not robust and cumbersome.
Better to not pass a transport but 
1. Create a future in MyProto
2. Set the future to some value (`None` for example) in `proto.connection_made()`
3. Wait for the future after `loop.sock_connect()` call

By this `prepare()` always return a connected server protocol, the change should eliminate race condition

**Review by asvetlov — CHANGES_REQUESTED** [review]

File: Lib/test/test_asyncio/test_unix_events.py (hunk 1 above)
Comment (pablogsal) [reply Lib/test/test_asyncio/test_unix_events.py]: Done in 1773bbd. Thanks for the suggestion! :)

**Comment by pablogsal** [comment]
I have made the requested changes; please review again

**Review by asvetlov — APPROVED** [review]
lgtm

**Pull request merged**